src/torchmetrics/classification/matthews_corrcoef.py

```diff
@@ -11,18 +11,216 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from typing import Any
+from typing import Any, Optional
 
 import torch
 from torch import Tensor
 
+from torchmetrics.classification import BinaryConfusionMatrix, MulticlassConfusionMatrix, MultilabelConfusionMatrix
 from torchmetrics.functional.classification.matthews_corrcoef import (
     _matthews_corrcoef_compute,
+    _matthews_corrcoef_reduce,
     _matthews_corrcoef_update,
 )
 from torchmetrics.metric import Metric
 
 
+class BinaryMatthewsCorrCoef(BinaryConfusionMatrix):
+    r"""Calculates `Matthews correlation coefficient`_ for binary tasks. This metric measures
+    the general correlation or quality of a classification.
+
+    Accepts the following input tensors:
+
+    - ``preds`` (int or float tensor): ``(N, ...)``. If preds is a floating point tensor with values outside
+      [0,1] range we consider the input to be logits and will auto apply sigmoid per element. Addtionally,
+      we convert to int tensor with thresholding using the value in ``threshold``.
+    - ``target`` (int tensor): ``(N, ...)``
+
+    Additional dimension ``...`` will be flattened into the batch dimension.
+
+    Args:
+        threshold: Threshold for transforming probability to binary (0,1) predictions
+        ignore_index:
+            Specifies a target value that is ignored and does not contribute to the metric calculation
+        normalize: Normalization mode for confusion matrix. Choose from:
+
+            - ``None`` or ``'none'``: no normalization (default)
+            - ``'true'``: normalization over the targets (most commonly used)
+            - ``'pred'``: normalization over the predictions
+            - ``'all'``: normalization over the whole matrix
+        validate_args: bool indicating if input arguments and tensors should be validated for correctness.
+            Set to ``False`` for faster computations.
+        kwargs: Additional keyword arguments, see :ref:`Metric kwargs` for more info.
+
+    Example (preds is int tensor):
+        >>> from torchmetrics import BinaryMatthewsCorrCoef
```

Comment:
```suggestion
```
I would tend not to import it from TM...
I know it nices in docs that people would copy past it, but it may have hidden issues as you are not sure that the imported class from TM is really this one especially when you import it from almost pkg root, not from this file...

Said so (I know it is not the case for this PR in particular but just caught my attention as all the rest is high-quality contributions as always from @SkafteNicki ) I would personally rather emphasize testing capabilities than simplicity in docs.. @Lightning-AI/core-metrics thought?

Reply:
would this be better:
```python
from torchmetrics.classification.fbeta import BinaryMatthewsCorrCoef
```
or is that also considered bad?

Reply:
that is probably the same as Will would complain that for showcasing the import path is too long and still, you need to while reviewing that the import is pointing to the file where example/class is... but lest leave it fo another discussion

Reply:
alrighty